api/mysagw/models.py

```diff
@@ -4,6 +4,48 @@
 from simple_history.models import HistoricalRecords
 
 
+class UniqueBooleanField(models.BooleanField):
+    """
+    BooleanField that guarantees for one True record and sets others to `False`.
+
+    By default only one record is allowed to be `True`. This can be further configured
+    by providing a list of field names as `together` param.
+
+    Example:
+        ```
+        >>> class Email(models.Model):
+        ...     user = models.ForeignKey(User)
+        ...     email = models.EmailField()
+        ...     default = UniqueBooleanField(together=["user"])
+        ```
+
+        This will enforce two things:
+         1. only allow for one entry where `default == True` per user
+         2. if only one record exists for a given user, it has `default = True`,
+            regardless of the input
+    """
+
+    def __init__(self, *args, together=None, **kwargs):
+        self.together = together if together else []
+        super().__init__(*args, **kwargs)
+
+    def pre_save(self, model_instance, add):
+        model = model_instance.__class__
+        together = model.objects.filter(
+            **{t: getattr(model_instance, t, None) for t in self.together},
```

Comment:
This will not work once someone puts a "dunder" notation reference in the `together` parameter (like `foo = UniqueBooleanField(together=["user__primary_group"])`). I don't think we have to worry about this now however :)

Reply:
True, but - as you wrote - that's something to worry about when such usecase arises.